src/coreclr/vm/methodtable.cpp

```diff
@@ -3841,6 +3841,18 @@ OBJECTREF MethodTable::Allocate()
     return AllocateObject(this);
 }
 
+// Boxes a value type using an input MethodTable*
+FCIMPL2(Object*, MethodTable::InternalBoxForMethodTable, MethodTable* pMethodTable, void* data)
+{
+    FCALL_CONTRACT;
+
+    _ASSERTE(pMethodTable != NULL);
+    _ASSERTE(data != NULL);
+
+    return OBJECTREFToObject(pMethodTable->Box(data));
```

Comment:
This needs to use the `HELPER_METHOD_FRAME` macro. You can omit the `HELPER_METHOD_FRAME` macro only if the FCall does not allocate or take locks. This allocates, so it needs `HELPER_METHOD_FRAME`.

Reply:
Ah, we were just wondering about that FCall needing some protection for GCs indeed.
Added in 23b4a3ffa26e331db50ddce66353e82ce7913722, hope I picked the right one 😅

Reply:
@jakobbotsch was also mentioning how the logic we need is also pretty much already there in `JIT_Box`, so whether we should duplicate it here or whether it may be better to just have the new box helper just HCCALL that one.

Reply:
Yes, referencing the existing `JIT_Box` helper in ecalllist.h should work fine.

Reply:
Ah, nice. Updated in f8cced6fbe4877adc33108d77217e818346c538d 😄